tidy3d/components/geometry/polyslab.py

```diff
@@ -1438,12 +1438,9 @@ def compute_derivative_vertices(self, derivative_info: DerivativeInfo) -> Traced
         edge_areas = edge_lengths
 
         # correction to edge area based on sidewall distance along slab axis
-        dim_axis = "xyz"[self.axis]
-        field_coords_axis = derivative_info.E_der_map[f"E{dim_axis}"].coords[dim_axis]
-        if len(field_coords_axis) > 1:
-            slab_height = abs(float(np.squeeze(np.diff(self.slab_bounds))))
-            if not np.isinf(slab_height):
-                edge_areas *= slab_height
+        slab_height = abs(float(np.squeeze(np.diff(self.slab_bounds))))
+        if not np.isinf(slab_height):
+            edge_areas *= slab_height
```

Comment:
Don't think I fully understand this one. So this would get ignored previously, or what was the issue?

Reply:
I dont think I fully understand it either, I think the idea was to ignore the slab thickness in some cases where the slab was considered infinitely thin (or the simulation was 2D). but it ended up never getting triggered in 3D because we changed the adjoint field monitor to be planar (in the middle of the slab bounds).